Add VPC support, disable master auto-provisioning, etc.

legal90 opened the pull request:
- Disable auto-provisioning of the master with a test job.
- Add VPC support. Instances will be created in the VPC specified by a variable.
- Fix provisioning of the master node. JRE 8 is required.
- Update defaults for variables, fix alignment.

pydavid commented on security_group.tf:
```diff
@@ -1,43 +1,44 @@
+  from_port   = 22
+  to_port     = 22
+  protocol    = "tcp"
   cidr_blocks = ["0.0.0.0/0"]
```
0.0.0.0/0 is too open. 
Can you put the public ip  ? `curl ifconfig.co`

pydavid commented on variables.tf:
```diff
@@ -1,50 +1,48 @@
   default = {
-    "us-east-1" = "ami-6869aa05"
-    "us-west-2" = "ami-7172b611"
+    "us-west-2" = "ami-39d39d41"
```
can you add a comment to say that it's the 2018 AWS official AMI ?

pydavid commented:
- Can you update the README.md 
-  Create a `main.tf`

legal90 commented:
@pydavid Thanks! I've done some refactoring, moved security groups out of this module's scope and added `main.tf` as a placeholder with a text reference to `master.tf` and `slave.tf` (there are no common resources for now).

Please, review it again
